pkg/controller/controller.go

```diff
@@ -512,6 +514,42 @@ func unmarshalParameters(in []byte) (map[string]interface{}, error) {
 	return parameters, nil
 }
 
+// fetchSecretParameters requests and returns the contents of the given secret as a map
+func fetchSecretParameters(kubeClient kubernetes.Interface, namespace string, secretRef *v1.LocalObjectReference) (map[string]interface{}, error) {
+	if kubeClient == nil {
+		return nil, errors.New("kubeClient is not defined")
+	}
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]interface{})
+	for k, v := range secret.Data {
+		var parameters interface{}
+		if json.Unmarshal(v, &parameters) != nil {
```

Comment:
> it could be a primitive value (int, string etc)

actually we can't support this with multiple Secret keys (as a flat map).
I have pointed to this limitation many times before, see https://github.com/kubernetes-incubator/service-catalog/issues/621#issuecomment-302833747.
You can't just say "I dunno, maybe it's JSON, or maybe it's a string", you have to choose one upfront in the code. We decided to choose JSON, and for types like `map[string]string` we should probably use single Secret key with JSON blob.